staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -184,19 +184,19 @@ func (e *eventBroadcasterImpl) recordToSink(event *eventsv1.Event, clock clock.C
 					Count:            1,
 					LastObservedTime: metav1.MicroTime{Time: clock.Now()},
 				}
-				return isomorphicEvent
+				// Make a copy to be able to continue updating the cache in parallel.
+				return isomorphicEvent.DeepCopy()
 			}
 			e.eventCache[eventKey] = eventCopy
-			return eventCopy
+			// Make a copy to be able to continue updating the cache in parallel.
+			return eventCopy.DeepCopy()
 		}()
 		if evToRecord != nil {
-			recordedEvent := e.attemptRecording(evToRecord)
-			if recordedEvent != nil {
-				recordedEventKey := getKey(recordedEvent)
-				e.mu.Lock()
-				defer e.mu.Unlock()
-				e.eventCache[recordedEventKey] = recordedEvent
-			}
+			// TODO: Add a metric counting the number of recording attempts
+			e.attemptRecording(evToRecord)
```

Comment:
nit: given you're no longer using the result of this function, let's change it to not return anything

Reply:
I was considering using the result in the future to differentiate successful attempts from failed attempts in a metric. Maybe I could make that clearer in the TODO and instead of returning an Event object I could just straight up return a boolean?